pennylane/templates/subroutines/arithmetic/temporary_and.py

```diff
@@ -262,7 +262,21 @@ def _temporary_and(wires: WiresLike, **kwargs):
 
 
 add_decomps(TemporaryAND, _temporary_and)
-# TODO: add add_decomps("Adjoint(TemporaryAND)", _adjoint_TemporaryAND) when MCMs supported by the pipeline
+
+
+def _adjoint_temporary_and_resources(*, base_class=None, base_params=None, **__):
+    return {ops.Hadamard: 1, ops.MidMeasure: 1, ops.Conditional: 1}
+
+
+@register_resources(_adjoint_temporary_and_resources)
+def _adjoint_TemporaryAND(wires: WiresLike, **kwargs):
+    r"""The implementation of adjoint TemporaryAND by mid-circuit measurements."""
```

Comment:
Could you please add a reference to the paper that originally proposed this decomposition here?
```suggestion
    r"""The implementation of adjoint TemporaryAND by mid-circuit measurements as found in https://arxiv.org/abs/1805.03662."""
```

Reply:
Could you also please point me to where in that paper this decomposition is defined?